hyundai_kia_connect_api/KiaUvoApiEU.py

```diff
@@ -396,281 +377,10 @@ def force_refresh_vehicle_state(self, token: Token, vehicle: Vehicle) -> None:
             else:
                 self._update_vehicle_drive_info(vehicle, state)
 
-    def _update_vehicle_properties_ccs2(self, vehicle: Vehicle, state: dict) -> None:
```

Comment:
Moved into Vehicle.py - the CCS2 schema seems largely the same, if not identical between Aus and EU so I'm opting to just have a shared implementation of the parser instead.

Reply:
It isn't used for the other regions and starts to break the structure.  I think we are better off creating a stamp based class that inherits for EU / AU.   So apiimpl -->  stampimpl --> kiaUvoAPIAU   this way you can override as required but most code would be in stampimpl.

Reply:
This is for vehicle state parsing, not stamps 🤔 

Happy to pull out to a separate class anyway, though, if it differs from NA. ooi, do we know if NA ccOS vehicles use this schema as well? or still different?

Reply:
Stamps is best way I can describe the api method.  Canada and USA are massively different api approach and parsing.

Reply:
Open to other names for this general api approach.  Just can't think of another name to classify the eu / au approach Kia uses.

Reply:
Ah gotcha. I'm aware that for existing vehicles, NA/CA have a different schema to EU/AU/CN, but I'm not sure if that's the case for the newer ccOS vehicles.
In EU/AU/CN, non-ccOS and ccOS vehicles _also_ have a different API, so I'm wondering if the new ccOS schema is just global now... but maybe that's just wishful thinking. It would be good to verify that though as it'd certainly make things more consistent on our side.

For now I'll go with what you've described and pull it out of vehicle 👍 other random names off the top of my head (none of them good either lol):
* Just call it EuAuCnVehicleParser.py vs NaCaVehicleParser.py, boring name but it's extremely obvious how they're different / where they're used 🤷 
* Just call it "type 1" or "type 2" or something like that, and have a comment on each one

Happy to go with whatever you prefer :)

Reply:
Maybe we just go with type 1?   CA, USA Hyundai, USA Kia are all different so they would remain as is.   Tough to name.   Reason I called it stamps is that CA, USA Hyundai and USA kia don't use stamps at all for authentication where as all the others do.  I did a quick draft of what I meant for structure I will create a PR.  Just purely as example. I could also merge it to build upon if it helps. Happy to rename to whatever works as well.

Reply:
sgtm! we can always rename it later if we come up with a better name as well :)

Reply:
In #530  I have renamed to ApiImplType1.  If it looks okay I can merge it and this PR can be changed to align with it?   I noticed China doesn't use stamps but rest seems to match AU/EU.   I don't think china works anymore though. Something changed and the person who added it hasn't come back to fix it.

Reply:
👌 happy to go with that, will update this PR, cheers

Reply:
I merged my PR and resolved the conflict here. I think last step would be moving the css2 code from vehicle back into the apiimpltype1 file.  Let me know once ready for next review.

Reply:
The above merge @cdnninja i think has caused a break - my HA is now showing this error:

`Setup failed for custom integration 'kia_uvo': Unable to import component: cannot import name 'stampApiImpl' from 'hyundai_kia_connect_api.ApiImplType1' (/usr/local/lib/python3.12/site-packages/hyundai_kia_connect_api/ApiImplType1.py)`